docs/concepts.md

```diff
@@ -6,22 +6,34 @@ menu: docs
 
 # Concepts
 
+## Application
+
+Any kind of piece of software that you would want to run in your Kubernetes cluster. It can be as simple as cleanup job or as complex as distributed system like Kafka.
+
 ## Operator
 
-High-level description of a deployable application. Contains metadata about the application (e.g., [Apache Kafka](https://github.com/kudobuilder/kudo/blob/master/config/samples/kafka-operator.yaml)).
+High-level description of a deployable application to be run in k8s cluster. Contains metadata about the application (e.g., [Apache Kafka](https://github.com/kudobuilder/kudo/blob/master/config/samples/kafka-operator.yaml)).
+You can have multiple versions of kafka ready to be installed in your cluster, all will belong to the same Operator.
 
 ## OperatorVersion
 
 Specific version of a deployable application, including configuration and lifecycle hooks for deployments, upgrades, and rollbacks (e.g., [Kafka version 2.4.0](https://github.com/kudobuilder/kudo/blob/master/config/samples/kafka-operatorversion.yaml)).
+This is already complete definition of application to be installed (except overridable parameters). By adding OperatorVersion to your cluster, no application is running yet.
 
 ## Instance
 
-Resource created to manage an instance of specific OperatorVersion. Instances are pets and have the same name throughout its entire lifecycle. (e.g., [Kafka 2.4.0 cluster with 1 broker](https://github.com/kudobuilder/kudo/blob/master/config/samples/kafka-instance.yaml)).
+When you create an instance, you provide missing parameters for installed OperatorVersion. Creating an instance typically causes bunch of services, pods or statefulsets to start running in your cluster.
+Instances have the same name throughout its entire lifecycle. (e.g., [Kafka 2.4.0 cluster with 1 broker](https://github.com/kudobuilder/kudo/blob/master/config/samples/kafka-instance.yaml)).
+
```

Comment:
Should we also mention that we add random strings https://github.com/kudobuilder/kudo/pull/437 ?

Reply:
do you think this is the right place? 🤔 not sure, but no strong opinion as well.